### PR TITLE
[AAA Dashboard] - Eligible Couples - Details Page

### PR DESCRIPTION
**Pull request by lwyszomi**
Hi @emord 

I have done the details page for Eligible Couples, I know that in the ICDS we have small problems with the performance so in this dashboard I decide to load each section asynchronously, so looking into this example:
eligible_couples_details

the first table (person, husband, other detail) have one ajax call, children details have one ajax call and the eligible couple details have one ajax call. Probably at the end, I will add a spinner to better represent that the data are loading.

Please let me know if you have any suggestions.

Regards,
Łukasz

### CODE REVIEW

**Comment by lwyszomi** [comment]
I can't find the correct icons which we have in the mockup, do you have access to them and you can share with me

File: custom/aaa/views.py [hunk 1]
```diff
@@ -279,3 +283,97 @@ def post(self, request, *args, **kwargs):
...
+        selected_year = int(self.request.POST.get('selectedYear', 0))
+        section = self.request.POST.get('section', '')
+        sub_section = self.request.POST.get('subsection', '')
+        beneficiary_id = self.request.POST.get('beneficiaryId', '')
+        data = {}
+        if section == 'child':
+            data = {}
+        elif section == 'pregnant_women':
+            data = {}
+        elif section == 'eligible_couples':
```
Comment (emord) [inline custom/aaa/views.py]: Can you add an `else` statement to this to send back 400 bad request?

File: custom/aaa/views.py [hunk 2]
```diff
@@ -195,23 +195,27 @@ def post(self, request, *args, **kwargs):
...
                     where=["NOT daterange(%s, %s) && any(pregnant_ranges)"],
                     params=[selected_date, selected_date + relativedelta(months=1)]
                 )
                 .values(
-                    'person_case_id', 'name', 'age',
+                    'id', 'name', 'age',
                     'currentFamilyPlanningMethod', 'adoptionDateOfFamilyPlaning')
             )[:10]
             data = list(data)
         elif beneficiary_type == 'pregnant_women':
```
Comment (emord) [inline custom/aaa/views.py]: similarly here, can you add an else to send back a 400

**Review by emord — APPROVED** [review]
Thanks for starting this.

I think most of the backend for all 3 reports should be availble through the models defined right now. For the next reports can you start including those in the details page? Its fine if you can't find all of them, you can put in constants if necessray

**Pull request merged**